### DISCUSSION TITLE
Creating a temperature field

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#16194] [Q&A General] [answered]
Hi everyone,

I want to create a temperature field (the central region is hot) as follows. 

![image](https://user-images.githubusercontent.com/6676821/99197359-c158a580-27dd-11eb-84c2-e297abec0968.png)

The temperature field is given by 
which from the paper (Hu, Acta Materialia 58 (2010) 3230–3237)
https://www.sciencedirect.com/science/article/pii/S1359645410000571

My question is that how can I build this temperature field in MOOSE? 
 
Best,

Peng

**Comment by WilkAndy** [comment, marked as answer]
I would use one of the "Parsed Function" objects in MOOSE.  For instance, if your temperature field is an IC, then use `FunctionIC`, while if your temperature field is an AuxVariable, then use a `ParsedAux` AuxKernel.   Just specify your function in the `function` input parameter in either case.

a

**Reply by dschwen** [reply]
Not quite. One way is to use a

```
[temperature]
  type = ParsedAux
  use_xyzt = true
  function = 'x^2 + cos(x ...`
  execute_on = INITIAL
[]
```

the other is to use

```
[temperature]
  type = FunctionAux
  function = 'x^2 + cos(x ...`
  execute_on = INITIAL
[]
```

No IC needed
